pkg/backend/telemetryapi/harvester_test.go

```diff
@@ -130,24 +139,28 @@ func TestHarvestCancelled(t *testing.T) {
 	})
 	h, _ := NewHarvester(func(cfg *Config) {
 		cfg.ErrorLogger = func(e map[string]interface{}) {
-			errs++
+			atomic.AddInt32(&errs, 1)
 		}
 		cfg.HarvestPeriod = 0
 		cfg.Client.Transport = rt
 		cfg.APIKey = "key"
+		cfg.Context = context.Background()
 	})
 	h.RecordSpan(Span{TraceID: "id", ID: "id"})
 
 	ctx, cancel := context.WithCancel(context.Background())
 	cancel()
 
 	h.HarvestNow(ctx)
+	wg.Wait()
 
-	if posts != 1 {
-		t.Error("incorrect number of tries tried", posts)
+	actualPosts := atomic.LoadInt32(&posts)
+	if actualPosts != 1 {
+		t.Error("incorrect number of tries tried", actualPosts)
 	}
-	if errs != 2 {
-		t.Error("incorrect number of errors logged", errs)
+	actualErrors := atomic.LoadInt32(&errs)
+	if actualErrors != 2 {
+		t.Error("incorrect number of errors logged", actualErrors)
```

Comment:
I would suggest to use assert for consistency, but I'm curious about `t.Error` vs `assert`.

Reply:
I wanted to do that but trying to keep the tests inside of `telemetryapi` consistent as one day we want to move these changes back to [Go Telemetry SDK](https://github.com/newrelic/newrelic-telemetry-sdk-go). They currently don't use [Testify](https://github.com/stretchr/testify).